Arrangement_on_surface_2/include/CGAL/Arr_geometry_traits/Bezier_x_monotone_2.h

```diff
@@ -1720,6 +1731,37 @@ bool _Bezier_x_monotone_2<RatKer, AlgKer, NtTrt, BndTrt>::_is_in_range
   return (false);
 }
 
+template <class RatKer, class AlgKer, class NtTrt, class BndTrt>
+bool _Bezier_x_monotone_2<RatKer, AlgKer, NtTrt, BndTrt>::_resultant_boundary_check
+    (const Algebraic& t, Bezier_cache& cache, bool is_min) const
+{
+  const Polynomial& t_resultant = cache.get_resultants().second;
+
+  Algebraic at_t = t_resultant.eval(t);
+  Algebraic at_boundary;
+  if (is_min)
+  {
+    Originator_iterator  s_org = _ps.get_originator (_curve, _xid);
+    at_boundary = t_resultant.eval(s_org->point_bound().t_min);
+  }
+  else
+  {
+    Originator_iterator  t_org = _pt.get_originator (_curve, _xid);
+    at_boundary = t_resultant.eval(t_org->point_bound().t_max);
+  }
+  at_t = at_t.abs();
+  at_boundary = at_boundary.abs();
+
+  if (CGAL::compare(at_t, at_boundary) == SMALLER)
+    return true;
```

Comment:
polynomial value check to see whether the boundary is closer to the real root than the calculated root